src/coreclr/interpreter/compiler.cpp

```diff
@@ -0,0 +1,583 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+#include "interpreter.h"
+
+#include "openum.h"
+
+static StackType g_stackTypeFromInterpType[] =
+{
+    StackTypeI4, // I1
+    StackTypeI4, // U1
+    StackTypeI4, // I2
+    StackTypeI4, // U2
+    StackTypeI4, // I4
+    StackTypeI8, // I8
+    StackTypeR4, // R4
+    StackTypeR8, // R8
+    StackTypeO,  // O
+    StackTypeVT  // VT
+};
+
+static InterpType g_interpTypeFromStackType[] =
+{
+    InterpTypeI4, // I4,
+    InterpTypeI8, // I8,
+    InterpTypeR4, // R4,
+    InterpTypeR8, // R8,
+    InterpTypeO,  // O,
+    InterpTypeVT, // VT,
+    InterpTypeI,  // MP,
+    InterpTypeI,  // F
+};
+
+// FIXME Use specific allocators for their intended purpose
+// Allocator for data that is kept alive throughout application execution,
+// being freed only if the associated method gets freed.
+void* InterpCompiler::AllocMethodData(size_t numBytes)
+{
+    return malloc(numBytes);
+}
+
+// Fast allocator for small chunks of memory that can be freed together when the
+// method compilation is finished.
+void* InterpCompiler::AllocMemPool(size_t numBytes)
+{
+    return malloc(numBytes);
+}
+
+// Allocator for potentially larger chunks of data, that we might want to free
+// eagerly, before method is finished compiling, to prevent excessive memory usage.
+void* InterpCompiler::AllocTemporary(size_t numBytes)
+{
+    return malloc(numBytes);
+}
+
+void* InterpCompiler::ReallocTemporary(void* ptr, size_t numBytes)
+{
+    return realloc(ptr, numBytes);
+}
+
+void InterpCompiler::FreeTemporary(void* ptr)
+{
+    free(ptr);
+}
+
+static int GetDataLen(int opcode)
+{
+    int length = g_interpOpLen[opcode];
+    int numSVars = g_interpOpDVars[opcode];
+    int numDVars = g_interpOpDVars[opcode];
```

Comment:
Is this correct? Both the numSVars and numDVars are set from the same array (g_interpOpDVars)